### ISSUE TITLE
`url` filter ignores `alt` parameter in paginated pages

### ISSUE THREAD

**Issue by eigengrau** [lektor/lektor#572]
The `url` filter ignores the `alt` parameter in some circumstances when paginated content is being generated.  I.e., on a non-primary alternative (say, `en`), of a paginated page, both the expressions `this | url(alt='de')` and `this | url(alt='_primary')` will evaluate to a path within the `en` alternative.

As a work-around, `this.record.path | url` will work, but will ignore the current page number being generated.

**Comment by ikus060** [comment]
I was about to report the exact same problem.

The defect seams to be in `db.py # get_virtual()`. The `alt` is not properly handled in this function and get lost.

https://github.com/lektor/lektor/pull/594

**Comment by eljoest** [comment]
This appears to be a duplicate of #216, could you please confirm if this would be resolved with PR #592?

**Comment by ikus060** [comment]
it's exactly the same fix. It's a duplicate.

**Comment by nixjdm** [comment]
@eljoest @ikus060 Thank you both for working on this and poking me to fix the tests and pull it in. I'll pick @ikus060's for the little nitpick that that is consistent with current argument positions. I consider this a bugfix, so I'm not going to wait on a test to be written. I've made a new issue for the tests if either of you would like to take a shot at it. #595

Thanks!

**Comment by nixjdm** [comment]
Fixed by #594

**Comment by nixjdm** [comment]
Thanks @eljoest
